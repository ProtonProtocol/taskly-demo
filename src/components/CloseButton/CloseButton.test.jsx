import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CloseButton from './CloseButton';
import CloseIcon from './CloseIcon';

const createTestProps = props => ({
  handleOnClick: jest.fn(),
  ...props,
});

describe('CloseButton component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<CloseButton {...props} />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<CloseButton {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('tests click', () => {
    const mockCallBack = jest.fn();

    const component = shallow(<CloseButton handleOnClick={mockCallBack} />);
    component.find('div').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});

describe('CloseIcon component', () => {
  it('renders without crashing', () => {
    shallow(<CloseIcon />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<CloseIcon />);
    expect(toJson(component)).toMatchSnapshot();
  });
});