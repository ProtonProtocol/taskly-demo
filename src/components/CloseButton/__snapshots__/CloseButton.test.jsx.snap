// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton component matches the rendered snapshot 1`] = `
<div
  className="close-button"
  onClick={[MockFunction]}
>
  <CloseIcon
    className=""
    fill="#000"
    opacity="1.0"
    viewBox="0 0 14 14"
    width="14px"
  />
</div>
`;
